crates/config/src/inline/natspec.rs

```diff
@@ -222,57 +226,85 @@ impl SolangParser {
+                                .as_str()
+                                .lines()
+                                .filter(|line| line.contains(INLINE_CONFIG_PREFIX))
+                                .map(|line| line.trim_start().trim_start_matches('*').trim())
+                                .collect::<Vec<_>>()
+                                .join("\n"),
+                        ),
+                    }
+                })
+                .join("\n")
         };
 
-        let mut prev_item_end = 0;
-        for item in &pt.0 {
-            let pt::SourceUnitPart::ContractDefinition(c) = item else {
-                prev_item_end = item.loc().end();
-                continue
-            };
-            let Some(id) = c.name.as_ref() else {
-                prev_item_end = item.loc().end();
-                continue
-            };
-            if id.name != contract_name {
-                prev_item_end = item.loc().end();
-                continue
-            };
-
-            // Handle doc comments in between the previous contract and the current one.
-            handle_docs(contract_id, None, prev_item_end, item.loc().start());
-
-            let mut prev_end = c.loc.start();
-            for part in &c.parts {
-                let pt::ContractPart::FunctionDefinition(f) = part else { continue };
-                let start = f.loc.start();
-                // Handle doc comments in between the previous function and the current one.
-                if let Some(name) = &f.name {
-                    handle_docs(contract_id, Some(name.name.as_str()), prev_end, start);
+        // Instantiate solar session
+        let sess = Session::builder()
+            .with_silent_emitter(Some("Inline config parsing failed".to_string()))
+            .build();
+
+        let _ = sess.enter(|| -> interface::Result<()> {
+            let arena = Arena::new();
+
+            let mut parser = Parser::from_source_code(
+                &sess,
+                &arena,
+                interface::source_map::FileName::Custom(contract_name.to_string()),
+                src.to_string(),
+            )?;
+
+            let source_unit = parser.parse_file().map_err(|e| e.emit())?;
+
+            for item in source_unit.items {
+                let ItemKind::Contract(ref c) = item.kind else { continue };
+
+                if c.name.as_str() != contract_name {
+                    continue
+                };
+
+                // Handle contract level doc comments
+                let docs = handle_docs(item);
+                if !docs.is_empty() {
+                    let docs_span = item.docs.iter().fold(Span::DUMMY, |s, doc| s.to(doc.span));
+                    let line = format!("{}:{}:0", docs_span.lo().0, docs_span.hi().0);
+                    natspecs.push(NatSpec {
+                        contract: contract_id.to_string(),
+                        function: None,
+                        line,
+                        docs,
```

Comment:
another thing is that `line` is documented and interpreted as `{line}:{column}:{length}`, but is in fact `{byte_start}:{byte_end}:{source_id}` in both parsers. This should be fixed in another PR